src/push-notifications.js

```diff
@@ -343,4 +363,14 @@ function isSupportedBrowser() {
   return isChrome || isOpera;
 }
 
+function isRegisteredToDifferentServerError(e) {
+  // This will only work in Chrome - we will need to test different browsers
```

Comment:
I don't think it's ideal to do this based on the error messages if they differ by browser. 

I had a look at the [Push API Spec](https://w3c.github.io/push-api/#pushmanager-interface). Apparently the error type will be `InvalidStateError` so we could check if `error.name === "InvalidStateError"`. But it looks like that type of error could also be returned if the "ActiveWorker" is null, so we might have to do extra checks to differentiate between those two scenarios.

Maybe we could just always call `_clearPushToken` before subscribing, rather than having to do it in response to an error.